src/framework/state/spec.ts

```diff
@@ -0,0 +1,139 @@
+import { CardID, Color, Spec } from "../types";
+
+export const getSpecStarters = (spec: Spec): Array<CardID> => {
+  const color = colorMap.get(spec);
+  if (color === undefined) {
+    throw new Error(`no color for spec ${spec}`);
+  }
+
+  const starters = colorStarters.get(color);
+  if (starters === undefined) {
+    throw new Error(`no starters for color ${color}`);
+  }
+
+  return starters;
+};
+
+const colorMap: Map<Spec, Color> = (() => {
```

Comment:
nit: The Map class in the STL in JS is actually kinda useless. It's much more normal to just use an object instead. To express this idea in typescript, normally you would use an annotation like `Record<Spec, Color>` (this is named `Record` specifically because `Map` was already taken, but is the more useful of the two)

Reply:
For most intents and purposes, are Records basically just Maps then? Like I should be able to replace my Maps with Records and few tweaks would be required beyond that?

Reply:
Record is much nicer than Map for this use case, as it turns out. Thanks for the suggestion.